internal/supervisor/import.go

```diff
@@ -0,0 +1,94 @@
+package supervisor
+
+import (
+	"time"
+
+	"github.com/mattermost/awat/model"
+	cloud "github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type ImportSupervisor struct {
+	logger log.FieldLogger
+	store  importStore
+	cloud  *cloud.Client
+}
+
+type importStore interface {
+	GetImportsInProgress() ([]*model.Import, error)
+	GetTranslation(id string) (*model.Translation, error)
+	UpdateImport(imp *model.Import) error
+}
+
+func NewImportSupervisor(store importStore, logger log.FieldLogger, provisionerURL string) *ImportSupervisor {
+	cloudClient := cloud.NewClient(provisionerURL)
+	return &ImportSupervisor{
+		logger: logger,
+		store:  store,
+		cloud:  cloudClient,
+	}
+}
+
+func (s *ImportSupervisor) Start() {
+	s.logger.Info("Import supervisor started")
+	go func() {
+		for {
+			err := s.supervise()
+			if err != nil {
+				s.logger.WithError(err).Error("failed an operation while supervising imports")
+			}
+			time.Sleep(60 * time.Second)
+		}
+	}()
+}
+
+func (s *ImportSupervisor) supervise() error {
+	imports, err := s.store.GetImportsInProgress()
```

Comment:
Isn't there a danger of race in case more than one import was requested?
I might be missing something but if the Provisioner can process only one import at a time (given we have one replica for processing imports) then the other installation will remain in `Stable` state until the first one is finished therefore the Import will be marked as succeeded after 10 seconds + other delays of supervisor runs etc.

Reply:
OK so my comment is a little bit incorrect and I think that's what's causing confusion here, and I'll change the wording to be clearer there, but also I will explain what's going on here and why there is not a race if more than one Import is requested.

When a Translation is completed, an Import is created automatically in state `import-requested` and with `StartAt == 0`.

When the Provisioner's ImportSupervisor wakes up, it requests 1 Import from the AWAT. The AWAT finds a new Import with state `import-requested`. It sets `StartAt`, locks the `Import` with an ID from the Provisioner, and moves the Import to `import-in-progress`, as one atomic operation. Then it returns the Import metadata to the Provisioner, for that Import.

Here's where the delay comes in: the Import is now locked on the AWAT side, but the Provisioner still needs to lock the Installation for the Import, so it has to look up the correct Installation, and lock it. If this fails, it fires off a request to the AWAT to unlock the Import [0] so that it can be retried in the future. My comment is confusing because I said the Provisioner "hasn't picked it up yet" and that's incorrect, the race that the delay prevents is one where the AWAT has locked the Import on the AWAT side but then the Provisioner fails to lock the Installation on the Provisioner side & transition the Installation state to `import-in-progress`. We only need to wait in order to make sure that the Provisioner has had the milliseconds that it needs to lock the Installation and transition the Installation's state, in case this supervisor's check runs between the change of the Import to `import-in-progress` on the AWAT side but before the `Installation` is transitioned to `import-in-progress`.

I will update the comment to reflect this. 

If there is another Import waiting, it will be in state `import-requested`. If another Provisioner pod, during the delay period, requests an Import, it will receive the Import in state `import-requested`, _not_ the Import that has already been locked by the first pod.

Likewise, this Supervisor will ignore any Imports waiting in `import-requested` state.

0: In explaining this logic, however, I have realized that the unlock method also needs to reset StartAt to 0, and will add that to this PR

Reply:
Also please let me know if you spot problems with or need clarification on any of that :point_up: 

This is a complicated enough synchronization mechanism / problem that I confused myself a couple of times while reasoning about the possible states, and if I'm overlooking some vast simplification to my approach that would make this all a bit clearer, I'm all for it.

Reply:
Also @gabrieljackson a sanity check from you as well, on the above, whenever you get around to reviewing this PR, would be appreciated